### ISSUE TITLE
atomWithStorage should not using default value if the value already exists in storage

### ISSUE THREAD

**Issue by sandiiarov** [pmndrs/jotai#542]
Atom with storage is using default value on mount and then it gets the value from storage.

```typescript
const darkModeAtom = atomWithStorage('darkMode', false);

const App = () => {
  const [darkMode, setDarkMode] = useAtom(darkModeAtom);
  console.log(darkMode); // it will output `false` then it will output storage value
  ...
```

Expected behavior:
atomWithStorage should return the storage value if it exists in storage
atomWithStorage should not use the default value if it exists in storage

Codesandbox with reproduction:
https://codesandbox.io/s/jotai-persistence-forked-g0j6g?file=/src/app.js

Steps to reproduce the issue:
Just refresh the page and observe the console

**Comment by dai-shi** [comment]
#540 would fix this. Can you confirm?

**Comment by dai-shi** [comment]
https://codesandbox.io/s/jotai-persistence-forked-klmfq?file=/src/app.js
Yes, it's fixed.

**Comment by mellson** [comment]
Nice, looking forward to this!

**Comment by mellson** [comment]
@dai-shi is this included in v1.0.1? Because I'm still getting a render with the default value before it loads the value from local storage
